### PR TITLE
Sorting Traffic Ops GET /api/*/cdns alphabetically

### LINKED ISSUE #3455: Traffic Ops GET /api/*/cdns should be sorted alphabetically by name by default
When clicking the snapshot button (the camera icon in the top right) in Traffic Portal, it shows the list of CDNs to choose from. Today, these CDNs appear to be sorted ascending by their numerical ID rather than alphabetically by their name. With a high number of CDNs to choose from, it would be easier to sift through the list alphabetically.

The CDN table (viewed by clicking the "CDNs" tab on the left) should also be sorted alphabetically by name. This appears to be a minor regression from the Perl TO API implementation which sorted by name by default (if no other `orderby` param is specified in the request URL): https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/lib/API/Cdn.pm#L34

### PR DESCRIPTION
**Pull request by rimashah25**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3455 

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

Login to TP UI and click on CDNs tab to ensure the CDNs are listed alphabetically
OR
Login to TP UI and click on Snapshot or Queue Server update and the drop down list should be sorted alphabetically
OR
Curl https://localhost:8443/api/3.0/cdns and the response list should be sorted alphabetically.

## If this is a bug fix, what versions of Traffic Control are affected?

master (7fa607e)

## The following criteria are ALL met by this PR

- [x]  This PR includes tests
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

PR testing can be done using API testing or visually looking at the TP UI (CDN tab or Snapshot tab)
Documentation for GET API calls is up to date.

### CODE REVIEW

**Review by rawlinp — CHANGES_REQUESTED** [review]
This seems kind of a roundabout way of providing a default sort for these various endpoints. Most of these support an `orderby` query parameter, and I think at least some of our endpoints will basically just choose a "default" `orderby` -- e.g. name, xmlID, etc. See traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go L999

**Review by rawlinp — CHANGES_REQUESTED** [review]
Ok, since these all use `GenericRead`, that previous advice is not super helpful.

For the handlers that call `GenericRead`, I think they can add the default `orderby` parameter to `APIInfo.Params` (if one isn't present already), before calling `GenericRead`. Looking at the code, that _should_ work.

**Comment by rimashah25** [comment]
> Ok, since these all use `GenericRead`, that previous advice is not super helpful.
> 
> For the handlers that call `GenericRead`, I think they can add the default `orderby` parameter to `APIInfo.Params` (if one isn't present already), before calling `GenericRead`. Looking at the code, that _should_ work.

Completed

**Comment by mitchell852** [comment]
should this PR be named something different since it's more than that?

**Comment by rawlinp** [comment]
It was more before, but we discussed the previous version offline and changed the approach. I believe @rimashah25 is going to add default sorting to the other API endpoints separately.

File: traffic_ops/traffic_ops_golang/cdn/cdns.go [hunk 1]
```diff
@@ -136,6 +137,8 @@ func (cdn *TOCDN) Read(h http.Header, useIMS bool) ([]interface{}, error, error,
 	// sorting alphabetically by cdn name.
 	if _, ok := cdn.APIInfo().Params["orderby"]; !ok {
 		cdn.APIInfo().Params["orderby"] = "name"
+	} else {
+		return nil, fmt.Errorf("unable to sort the list"), nil, http.StatusInternalServerError, nil
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/cdn/cdns.go]: Doesn't this mean that if I try to set `orderby` myself in the URL it'll give me back an internal server error?

Comment (rimashah25) [reply traffic_ops/traffic_ops_golang/cdn/cdns.go]: Resolved conversation based on offline chat.

File: traffic_ops/testing/api/v3/cdns_test.go [hunk 2]
```diff
@@ -92,6 +93,16 @@ func CreateTestCDNs(t *testing.T) {
 
 }
 
+func SortTestCDNs(t *testing.T) {
+	var header http.Header
+	for _, cdn := range testData.CDNs {
+		_, reqInf, err := TOSession.GetCDNByNameWithHdr(cdn.Name, header)
```
Comment (rawlinp) [inline traffic_ops/testing/api/v3/cdns_test.go:99]: So this doesn't really verify that the response is sorted alphabetically. Maybe we should take the response of all the CDNs, iterate over them, and use `strings.Compare` to make sure the response is sorted as expected.

Comment (rimashah25) [reply traffic_ops/testing/api/v3/cdns_test.go]: Fixed test case.

**Comment by rawlinp** [comment]
Since we can't reorder the existing CDNs in tc-fixtures.json, can we add a _new_ CDN named something like `a-cdn` on the end?

**Comment by rimashah25** [comment]
> Since we can't reorder the existing CDNs in tc-fixtures.json, can we add a _new_ CDN named something like `a-cdn` on the end?

I can try.. but messing with tc_fixtures is kinda harrowing experience.

**Review by rawlinp — APPROVED** [review]
LGTM, checks passed

**Pull request merged**